ambry-api/src/main/java/com/github/ambry/router/Router.java

```diff
@@ -34,10 +35,12 @@
    * @param blobId The ID of the blob for which blob data is requested.
    * @param options The options associated with the request. This cannot be null.
    * @param callback The callback which will be invoked on the completion of the request.
+   * @param quotaChargeEventListener Listener interface to charge quota cost for the operation.
    * @return A future that would eventually contain a {@link GetBlobResult} that can contain either
    *         the {@link BlobInfo}, the {@link ReadableStreamChannel} containing the blob data, or both.
    */
-  Future<GetBlobResult> getBlob(String blobId, GetBlobOptions options, Callback<GetBlobResult> callback);
+  Future<GetBlobResult> getBlob(String blobId, GetBlobOptions options, Callback<GetBlobResult> callback,
```

Comment:
I think we should create a different method for this one, since there are some other callers who don't want to use a eventListener here.

Reply:
We can either create a different method, or pass null or empty lambda when callers don't want to use it. Keeping just one method helps keep Router less complicated in my opinion.
Still if you think otherwise, I can fix it.

Reply:
Have we considered adding `quotaChargeEventListener` as a `GetBlobOption`/`PutBlobOption`? These have associated builders that can help with optional arguments like this.

We could create options classes for delete/undelete/ttlupdate to make it symmetric. Alternatively, do those methods even need to be passed a listener as opposed to the frontend handling this logic in the callback.

Reply:
Putting the listener or callback in Options class for each operation will make the APIs cleaner. We can take it as a follow up item after this PR is merged.